src/server/pfs/server/driver_v2.go

```diff
@@ -961,3 +970,45 @@ func (d *driverV2) deleteCommit(txnCtx *txnenv.TransactionContext, userCommit *p
 
 	return nil
 }
+
+func (d *driverV2) createTmpFileSet(server pfs.API_CreateTmpFileSetServer) (string, error) {
+	ctx := server.Context()
+	opts := []fileset.UnorderedWriterOption{fileset.WithWriterOption(fileset.WithTTL(30 * time.Minute))}
+	id := uuid.NewWithoutDashes()
+	if err := d.withUnorderedWriter(ctx, tmpRepo, id, func(uw *fileset.UnorderedWriter) error {
+		req := &pfs.PutTarRequestV2{
+			Tag: "",
+		}
+		_, err := putTar(uw, server, req)
+		return err
+	}, opts...); err != nil {
+		return "", err
+	}
+	return id, nil
+}
+
+func (d *driverV2) renewTmpFileSet(ctx context.Context, id string, ttl time.Duration) error {
+	if ttl > maxTTL {
+		return errors.Errorf("ttl (%d) exceeds max ttl (%d)", ttl, maxTTL)
+	}
+	// check that it is the correct length, to prevent malicious renewing of multiple filesets
+	// len(hex(uuid)) == 32
+	if len(id) != 32 {
+		return errors.Errorf("invalid id (%s)", id)
+	}
+	p := path.Join(tmpRepo, id)
+	_, err := d.storage.RenewFileSet(ctx, p, ttl)
+	return err
+}
+
+func (d *driverV2) inspectCommit(pachClient *client.APIClient, commit *pfs.Commit, blockState pfs.CommitState) (*pfs.CommitInfo, error) {
+	if commit.GetRepo().GetName() == tmpRepo {
```

Comment:
What is going on here? Why do we return a commit info, shouldn't we just error?

Reply:
We want to call inspect commit on all the accessors before proceeding.  This creates a phony commit info so the accessors can proceed as usually without any changes.